frontend/src/stores/searchParams/index.ts

```diff
@@ -92,23 +97,69 @@ export const useSimpleSearchParamsStore = defineStore(
     const setSort = (value: string) => (sort.value = value);
 
     function $reset() {
-      const initialState = getInitialState(router);
+      reinitializeFromQuery(route.query);
+    }
+
+    function reinitializeFromQuery(routerQuery: LocationQuery) {
+      const initialState = getInitialState(routerQuery);
 
       query.value = initialState.query;
       pageNumber.value = initialState.pageNumber;
       category.value = initialState.category;
       itemsPerPage.value = initialState.itemsPerPage;
       sort.value = initialState.sort;
       court.value = initialState.court;
-      dateParams.$reset();
+      dateParams.reset(initialState);
     }
 
     /*
```

Comment:
If I were you I'd get rid of the extra comments, usually according to clean code principles, a code should have structure and names that are self explanatory and wouldn't need comments to illustrate what it does. That's the case in other files too, but I thought to leave the comment here anyhow for you to consider if you like.

Reply:
Clean code doesn't forbid all comments ;) I would keep a comment detailing that the intention is to prevent infinite loops.
Your suggestion does make sense though!